File: ChessAPI/Domain/Entities/Match.cs
using Domain.Enums;

namespace Domain.Entities;

public class Match
{
    public int Id { get; set; }
    public int TournamentId { get; set; }
    public int Round { get; set; }
    public int Player1Id { get; set; }
    public int Player2Id { get; set; }
    public MatchResult Result { get; set; } = MatchResult.Pending;
    
    // Propriétés de navigation optionnelles
    public Player? Player1 { get; set; }
    public Player? Player2 { get; set; }
}

File: ChessClient/src/app/shared/models/match.ts
export interface Match {
    id: number;
    player1Id: number;
    player2Id: number;
    tournamentId: number;
    round: number;
    result: MatchResult;
}

export enum MatchResult {
    Pending = 0,
    WhiteWin = 1,
    BlackWin = 2,
    Draw = 3
}